metadata_fetcher/fetchers/pra_fetcher.py

```diff
@@ -0,0 +1,159 @@
+import json
+from xml.etree import ElementTree
+from .Fetcher import Fetcher
+from requests.auth import HTTPBasicAuth
+import requests
+import re
+
+
+class PraFetcher(Fetcher):
+
+    BASE_URL = "https://us.preservica.com/api/entity/v6.0"
+
+    NAMESPACES = {"pra": "http://preservica.com/EntityAPI/v6.0"}
+
+    def __init__(self, params):
+        super(PraFetcher, self).__init__(params)
+
+        # If `next_url` is a param, we know that this is not
+        # the fetch of the first page, so skip setting those
+        # attributes
+        if "next_url" in params:
+            for key in params:
+                setattr(self, key, params[key])
+            return
+
+        credentials = params.get("harvest_data").get("harvest_extra_data")
+        self.basic_auth_credentials = ([v.strip() for v in credentials.split(',')])
+        self.internal_collection_id = re.search(r"(?<=SO_)[0-9a-f-]+",
+                                                params.get("harvest_data").get("url")).group(0)
+        self.original_url = f"{self.BASE_URL}/structural-objects/{self.internal_collection_id}/children"
+        self.next_url = self.get_first_page_url()
+
+    def get_first_page_url(self):
+        """
+        Two possibilities exist:
+
+        1) The `original_url` contains a list of IO children and is the first page of results, or
+        2) The `original_url` is a list of SO children, hopefully contain 1 item, and we must do more
+           to get to the list of IO children:
+
+           Fetching the first page of items requires two requests to get to it. The first, the original_url,
+           returns a ChildrenResponse, from which the URL from the first Child is extracted. The second request returns
+           an EntityResponse, from which the URL is extracted from AdditionalInformation/Children's text node.
+        """
+        request = self.build_url_request(self.original_url)
+        response = requests.get(**request)
+        root = ElementTree.fromstring(response.text)
+
+        # If we have IO (Information Object) children, then this is the first page. Otherwise,
+        # we have to continue digging.
+        io_children = root.findall(".//pra:Child[@type='IO']", self.NAMESPACES)
+        if len(io_children) > 0:
+            return self.original_url
+
+        child_url = root.find(".//pra:Child[@type='SO']", self.NAMESPACES).text
+
+        request = self.build_url_request(child_url)
+        response = requests.get(**request)
+        root = ElementTree.fromstring(response.text)
+
+        return root.find(".//pra:AdditionalInformation/pra:Children", self.NAMESPACES).text
```

Comment:
Further discovery is required, but the `original_url`, the first URL generated from the internal collection id seems to come in at least two forms, which can be see in looking at this URL for collections:

26782: `https://us.preservica.com/api/entity/v6.0/structural-objects/079822c0-b1fc-45df-8166-19880b12edba/children`

26460: `https://us.preservica.com/api/entity/v6.0/structural-objects/fd56c609-a9aa-4e8e-8745-60d1128befe0/children`

The first one lists children with the type of IO, information objects. This is a page of results, with the <Child> elements referencing the objects and their metadata.

The second lists a single child of the type SO, a structural object, which requires an additional request to get to the paginated list of items.

The fetcher currently works for both these collections, but needs to be tested against others.